### ISSUE TITLE
[ColecoVision] - Submerged turtles show a green box for a few frames in Frogger

### ISSUE THREAD

**Issue by remutro** [ares-emulator/ares#84]
I know this is already known within the team, but it is probably best to create an issue to track it at this point. This is the only known corruption in ColecoVision games that I am aware of. Turtles on the bottom row will show green squares for a few frames while going down/and coming up. It doesn't seem to occur on the turtles on the top row. This occurs in Ares v123. Here is a screenshot:

![Frogger](https://user-images.githubusercontent.com/84990500/130367469-82782f6a-7615-4863-bf2a-fa2651b41e10.png)

**Comment by invertego** [comment]
I took a look, and unfortunately this is a game bug. The game is inadvertently messing up its own copies to VRAM if a vblank interrupt comes in at just the wrong time. Here's an instruction trace that shows it happening:

```
1d08  ld   a,l
1d09  out  ($bf),a      ; write low address byte to port

CPU Interrupt: NMI
0066  jp   $8021
8021  push af
8022  call $1fdc
1fdc  jp   $1d57
1d57  in   a,($bf)      ; resets target of next port write
1d59  ret
8025  pop  af
8026  retn

1d0b  ld   a,h
1d0c  out  ($bf),a      ; attempt to write high address byte to port (writes low byte instead)
```

I also found [a report of similar issues](https://atariage.com/forums/topic/278887-does-anyone-notice-the-odd-graphical-glitch-on-frogger/) manifesting on real hardware, though that person was also using an SD cart adapter which could be responsible for disrupting the timing of the initial vblank. I don't know if this issue would reproduce on real hardware with a standard cart after enough attempts, or if the timing of the first vblank is consistent enough to avoid the bug entirely.

There are multiple ways this could be addressed in ares, but choosing one will require some thought.
